### PR TITLE
feat: ProductGallery — scroll-snap slider + dialog zoom

### PR DESCRIPTION
**Pull request by spokospace**
## Summary
- New `ProductGallery` component replacing Swiper + LightGallery with a custom CSS scroll-snap solution (~2-3 KB vs ~68 KB)
- Includes: main slider, thumbnails, fullscreen `<dialog>` with zoom/pan, arrow navigation, fraction counter
- Full accessibility: `aria-roledescription="carousel"`, `aria-pressed` on thumbnail buttons, `aria-live` counters, `prefers-reduced-motion` support
- Shared component for catalog.polo.blue and sale.polo.blue via subpath import

## Test plan
- [ ] Verify slider navigation: drag, arrows, thumbnails, keyboard
- [ ] Test fullscreen dialog: open on click, zoom on click/wheel/double-tap, pan on mousemove
- [ ] Check mobile: swipe, counter, no thumbnails, double-tap zoom
- [ ] Verify edge cases: single image, empty state, many images (10+)
- [ ] Test in catalog.polo.blue and sale.polo.blue integration

## Summary by CodeRabbit

* **New Features**
  * Introduced a product gallery: CSS scroll‑snap main slider, synchronized thumbnail strip (desktop), slide counters, desktop arrows, fullscreen lightbox with zoom & pan, keyboard & arrow navigation, drag‑to‑scroll, thumbnail syncing, and an empty‑state view.

* **Documentation**
  * Added comprehensive usage docs and examples for the new product gallery component.

### CODE REVIEW

File: src/components/Product/product-gallery.ts [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+    // Sync main slider to dialog position
+    scrollToSlide(slider, dialogActiveIndex);
+  }
+
+  dialogClose?.addEventListener('click', closeDialog);
+
+  // Close on backdrop click
+  dialog.addEventListener('click', e => {
+    if (e.target === dialog) closeDialog();
+  });
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:243]: Fixed in 6134f50. Extracted teardown into a separate \ function and added \ so native Escape triggers the same cleanup (zoom reset, observer disconnect, body overflow restore, main slider sync).

File: src/components/Product/product-gallery.ts [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+    document.body.style.overflow = 'hidden';
+
+    // Preload full-res images around the starting slide
+    preloadDialogImages(startIndex);
+
+    // Scroll to the clicked image (instant, no animation)
+    const targetSlide = dialogSlider.querySelector<HTMLElement>(`[data-index="${startIndex}"]`);
+    if (targetSlide) {
+      targetSlide.scrollIntoView({ behavior: 'instant' as ScrollBehavior, inline: 'center' });
+    }
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:196]: Fixed in 6134f50. Changed \ to \ which is the standard value for instant scrolling.

File: src/components/Product/product-gallery.ts [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+      }
+    }
+  }
+
+  function openDialog(startIndex: number) {
+    if (!dialog || !dialogSlider) return;
+
+    dialog.showModal();
+    document.body.style.overflow = 'hidden';
+
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:188]: Fixed in 6134f50. Now saves the previous `body.style.overflow` value before setting `hidden` and restores it on close.

File: src/components/Product/product-gallery.ts [hunk 4]
```diff
@@ -0,0 +1,500 @@
+/* eslint-disable no-undef */
+/**
+ * ProductGallery — scroll-snap slider + dialog lightbox
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts]: Fixed in 6134f50. Configured `globals.browser` in ESLint flat config for `.ts` files, so all browser DOM globals are properly declared. Removed the blanket `eslint-disable no-undef`.

File: src/components/Product/ProductGallery.astro [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+              <button
+                type="button"
+                class:list={[
+                  'pg-thumb shrink-0 h-20 w-auto p-0 border-2 border-transparent rounded-sm cursor-pointer bg-transparent overflow-hidden',
+                  { 'is-active': index === 0 },
+                ]}
+                data-thumb-index={index}
+                role="tab"
+                aria-selected={index === 0 ? 'true' : 'false'}
+                aria-label={`Image ${index + 1}`}
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro]: Fixed in 6134f50. Removed `role="tablist"` and `role="tab"` from both main and dialog thumbnail strips. Using plain buttons with `aria-pressed` instead, which accurately reflects the toggle-like selection behavior without requiring the full ARIA tabs pattern.

File: src/components/Product/ProductGallery.astro [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+                <button
+                  type="button"
+                  class:list={[
+                    'pg-dialog-thumb shrink-0 h-15 w-auto p-0 border-2 border-transparent rounded-sm cursor-pointer bg-transparent overflow-hidden',
+                    { 'is-active': index === 0 },
+                  ]}
+                  data-thumb-index={index}
+                  role="tab"
+                  aria-selected={index === 0 ? 'true' : 'false'}
+                  aria-label={`Image ${index + 1}`}
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro]: Fixed in 6134f50 — same approach as main thumbnails: removed tab roles, using `aria-pressed` on plain buttons.

File: src/components/Product/ProductGallery.astro [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+            <span data-gallery-current>1</span>/{total}
+          </div>
+
+          <button
+            type="button"
+            class="pg-arrow hidden md:flex absolute top-50% left-0 z-10 w-10 h-10 items-center justify-center border-none bg-transparent text-primary-400 cursor-pointer drop-shadow-md"
+            data-gallery-prev
+            data-dir="prev"
+            aria-label="Previous image"
+            disabled
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro:97]: Fixed in 6134f50. Changed `top-50%` to `top-1/2` which is the correct UnoCSS fractional syntax.

File: src/components/Product/ProductGallery.astro [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+          >
+            {images.map((image, index) => (
+              <div
+                class="pg-slide w-full min-w-full overflow-hidden md:cursor-zoom-in"
+                role="group"
+                aria-roledescription="slide"
+                aria-label={`${index + 1} / ${total}`}
+                data-index={index}
+                data-full={image.full || image.src}
+              >
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro:68]: Fixed in 6134f50. Added `tabindex="0"` to each slide div and keyboard handlers for Enter/Space to open the dialog.

File: src/components/Product/ProductGallery.astro [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+              stroke-width="2.5"
+              aria-hidden="true"
+            >
+              <polyline points="15 18 9 12 15 6" />
+            </svg>
+          </button>
+          <button
+            type="button"
+            class="pg-dialog-arrow hidden md:flex absolute top-50% right-3 -translate-y-50% z-10 w-12 h-12 items-center justify-center border-none bg-black/45 text-neutral-400 cursor-pointer rounded transition-colors"
+            data-dialog-next
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro:247]: Fixed in 6134f50. Changed `top-50%` to `top-1/2` and `-translate-y-50%` to `-translate-y-1/2` for both dialog arrows.

File: .astro/types.d.ts [hunk 10]
```diff
@@ -1,2 +1 @@
 /// <reference types="astro/client" />
```
[the remark this answers is not in the document]
Comment (spokospace) [reply .astro/types.d.ts:1]: The `content.d.ts` reference is still present in `.astro/types.d.ts` — this was auto-generated by Astro and the reference path was only normalized (relative path format), not removed. No action needed.

File: src/components/Product/product-gallery.css [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+.pg-main:hover {
+  .pg-arrow[data-dir='prev']:not(:disabled) {
+    transform: translate(50%, -50%);
+  }
+  .pg-arrow[data-dir='next']:not(:disabled) {
+    transform: translate(-50%, -50%);
+  }
+}
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.css:59]: Fixed in 6134f50. Added `.pg-main:focus-within` alongside `.pg-main:hover` so arrow buttons are revealed on keyboard focus too.

File: src/components/Product/product-gallery.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:243]: Fixed in 6134f50. Implemented the suggested pattern: extracted teardown into a separate function and added \. Also saves/restores original body overflow.

File: src/components/Product/ProductGallery.astro (hunk 5 above)
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro]: Fixed in 6134f50. Chose the simpler approach: removed tab roles entirely. Thumbnails are now plain buttons with `aria-pressed` for active state.

File: src/components/Product/ProductGallery.astro [hunk 12]
```diff
@@ -0,0 +1,323 @@
...
+                aria-roledescription="slide"
+                aria-label={`${index + 1} / ${total}`}
+                data-index={index}
+              >
+                <div class="pg-zoom" data-zoom-container>
+                  <img
+                    src={image.full || image.src}
+                    alt={image.alt}
+                    loading={index === 0 ? 'eager' : 'lazy'}
+                    decoding={index === 0 ? 'sync' : 'async'}
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro]: Fixed in 6134f50. All dialog images now use `loading="lazy"` and `decoding="async"`. The `preloadDialogImages()` function handles fast-open by switching to eager on demand.

File: src/pages/components/product-gallery.mdx [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+</div>
+
+## Props
+
+| Prop | Type | Default | Description |
+|------|------|---------|-------------|
+| `images` | `GalleryImage[]` | `[]` | Array of image objects |
+| `aspectRatio` | `string` | `"4/3"` | CSS aspect-ratio for the slider |
+| `class` | `string` | — | Additional CSS class on root element |
+
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/pages/components/product-gallery.mdx:134]: Fixed in 6134f50. Added `ariaLabel` row to the props table with default `"Product image gallery"`.

File: src/components/Product/product-gallery.ts [hunk 14]
```diff
@@ -0,0 +1,513 @@
...
+    isDown = false;
+
+    if (pointerId !== null) {
+      try {
+        container.releasePointerCapture(pointerId);
+      } catch {
+        /* pointer may already be released */
+      }
+      pointerId = null;
+    }
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:365]: Fixed in 6134f50. Moved `setPointerCapture` to the `pointerdown` handler (wrapped in try/catch) and removed the duplicate call from `pointermove`.

File: src/components/Product/product-gallery.ts [hunk 15]
```diff
@@ -0,0 +1,513 @@
...
+  });
+
+  container.addEventListener('click', e => {
+    if (isZoomed) {
+      zoomOut();
+    } else {
+      zoomIn(e.clientX, e.clientY);
+      container.addEventListener('mousemove', onMouseMove);
+    }
+  });
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:514]: Fixed in 6134f50. Added a touch timestamp guard — records `lastTouchTime` on `touchstart`/`touchend` and skips the click handler if within 500ms of a touch event. Double-tap zoom on mobile is unaffected.

File: src/components/Product/product-gallery.css [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+/* ── Zoom (native overflow scroll, no crop) ── */
+
+.pg-zoom {
+  @apply flex items-center justify-center overflow-hidden cursor-zoom-in;
+  width: 100%;
+  height: 100%;
+
+  &.is-zoomed {
+    @apply cursor-zoom-out;
+  }
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.css:95]: Fixed in 890f281. Added `cursor-zoom-out` to UnoCSS theme in `uno-config/theme/effects.ts` so the `@apply cursor-zoom-out` resolves correctly.

File: src/components/Product/ProductGallery.astro [hunk 17]
```diff
@@ -0,0 +1,320 @@
...
+            {images.map((image, index) => (
+              <div
+                class="pg-slide w-full min-w-full overflow-hidden md:cursor-zoom-in"
+                role="group"
+                aria-roledescription="slide"
+                aria-label={`${index + 1} / ${total}`}
+                data-index={index}
+                data-full={image.full || image.src}
+                tabindex="0"
+              >
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro:68]: Fixed in 0795a2c. Changed slide divs from `role="group"` to `role="button"` and added `aria-haspopup="dialog"` so screen readers announce them as actionable controls that open a dialog.

File: src/components/Product/product-gallery.css [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+  &[open] {
+    @apply flex flex-col;
+  }
+}
+
+/* ── Zoom (native overflow scroll, no crop) ── */
+
+.pg-zoom {
+  @apply flex items-center justify-center overflow-hidden cursor-zoom-in;
+  width: 100%;
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.css]: Fixed in 0795a2c. Updated the comment to "Zoom (transform-origin based, GPU-accelerated)" which accurately describes the implementation.

File: eslint.config.js [hunk 19]
```diff
@@ -3,6 +3,7 @@ import typescript from '@typescript-eslint/eslint-plugin';
 import typescriptParser from '@typescript-eslint/parser';
 import vue from 'eslint-plugin-vue';
 import astro from 'eslint-plugin-astro';
+import globals from 'globals';
```
[the remark this answers is not in the document]
Comment (spokospace) [reply eslint.config.js:6]: Fixed in 0795a2c. Added `globals` to `devDependencies` via `pnpm add -D globals`.

File: src/components/Product/ProductGallery.astro [hunk 20]
```diff
@@ -0,0 +1,321 @@
...
+                aria-label={`${index + 1} / ${total}`}
+                aria-haspopup="dialog"
+                data-index={index}
+                data-full={image.full || image.src}
+                tabindex="0"
+              >
+                <img
+                  src={image.src}
+                  alt={image.alt}
+                  loading={index === 0 ? 'eager' : 'lazy'}
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro:72]: Fixed in c47945b. All `aria-label`s now include `image.alt` for better screen reader context:

- Main slides: `"Open image 1 of 5: {image.alt}"`
- Dialog slides: `"Image 1 of 5: {image.alt}"`
- Main thumbnails: `"Image 1: {image.alt}"`
- Dialog thumbnails: `"Image 1: {image.alt}"`

**Comment by spokospace** [comment]
**Re: CodeRabbit nitpick — `prefers-reduced-motion` (product-gallery.css lines 8-12)**

Fixed in c47945b. Added a `@media (prefers-reduced-motion: reduce)` block that:
- Sets `scroll-behavior: auto` on `.pg-slider` and `.pg-dialog-slider` (instant snap instead of smooth)
- Disables `transition` on `.pg-arrow` and `.pg-zoom-img` (no transform animations)

File: src/components/Product/product-gallery.ts [hunk 21]
```diff
@@ -0,0 +1,547 @@
...
+
+  function createSlideObserver(container: HTMLElement, onActiveChange: (index: number) => void) {
+    const observer = new IntersectionObserver(
+      entries => {
+        for (const entry of entries) {
+          if (entry.isIntersecting && entry.intersectionRatio >= 0.5) {
+            const index = Number((entry.target as HTMLElement).dataset.index);
+            if (!isNaN(index)) onActiveChange(index);
+          }
+        }
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:51]: Fixed in 56edbb1. The observer callback now picks the single entry with the highest `intersectionRatio` instead of firing `onActiveChange` for every intersecting entry, preventing flicker mid-drag.

File: src/components/Product/product-gallery.ts [hunk 22]
```diff
@@ -0,0 +1,547 @@
...
+    }
+  }
+
+  function openDialog(startIndex: number) {
+    if (!dialog || !dialogSlider) return;
+
+    dialog.showModal();
+    savedOverflow = document.body.style.overflow;
+    document.body.style.overflow = 'hidden';
+
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:188]: Fixed in 56edbb1. Added `dialog.open` guard in `openDialog()` — `if (!dialog || !dialogSlider || dialog.open) return;` — so double-clicks can't throw `InvalidStateError`.

File: src/components/Product/product-gallery.ts [hunk 23]
```diff
@@ -0,0 +1,547 @@
...
+    isZoomed = true;
+    container.classList.add('is-zoomed');
+
+    const rect = img.getBoundingClientRect();
+    // Set transform-origin to click point (relative to image)
+    const originX = ((clientX - rect.left) / rect.width) * 100;
+    const originY = ((clientY - rect.top) / rect.height) * 100;
+    img.style.transformOrigin = `${originX}% ${originY}%`;
+    img.style.transform = `scale(${ZOOM_SCALE})`;
+  }
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts:463]: Fixed in 56edbb1. Added a `clamp(value, 0, 100)` helper and applied it to both `originX` and `originY` in `zoomIn()`, so clicking outside the rendered image area no longer produces out-of-range origins.

File: src/components/Product/product-gallery.ts [hunk 24]
```diff
@@ -0,0 +1,547 @@
...
+    container.removeEventListener('mousemove', onMouseMove);
+  }
+
+  // Mouse-move panning: move the transform-origin proportionally to cursor position
+  function onMouseMove(e: MouseEvent) {
+    if (!isZoomed) return;
+    const containerRect = container.getBoundingClientRect();
+    const x = ((e.clientX - containerRect.left) / containerRect.width) * 100;
+    const y = ((e.clientY - containerRect.top) / containerRect.height) * 100;
+    img.style.transformOrigin = `${x}% ${y}%`;
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/product-gallery.ts]: Fixed in 56edbb1. `onMouseMove` now uses `img.getBoundingClientRect()` instead of the container rect, and clamps percentages to 0–100%. This ensures accurate panning even when the image is letterboxed.

File: src/components/Product/ProductGallery.astro [hunk 25]
```diff
@@ -0,0 +1,321 @@
...
+              <polyline points="15 18 9 12 15 6" />
+            </svg>
+          </button>
+          <button
+            type="button"
+            class="pg-arrow hidden md:flex absolute top-1/2 right-0 z-10 w-10 h-10 items-center justify-center border-none bg-transparent text-primary-400 cursor-pointer drop-shadow-md"
+            data-gallery-next
+            data-dir="next"
+            aria-label="Next image"
+          >
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/components/Product/ProductGallery.astro:118]: Fixed in 56edbb1. The "Next image" button now has `disabled={total <= 1}` so it's correctly disabled server-side for single-image galleries.

File: eslint.config.js [hunk 26]
```diff
@@ -17,9 +18,8 @@ export default [
         sourceType: 'module',
       },
       globals: {
-        console: 'readonly',
-        window: 'readonly',
-        document: 'readonly',
+        ...globals.browser,
+        NodeListOf: 'readonly',
       },
```
[the remark this answers is not in the document]
Comment (spokospace) [reply eslint.config.js:22]: Fixed in 56edbb1. Removed `NodeListOf` from globals and instead disabled the base `no-undef` rule for TS files (`'no-undef': 'off'`). TypeScript's own type checker handles undefined-variable checks natively, making the ESLint rule redundant and potentially misleading for type-only identifiers.

File: src/pages/components/product-gallery.mdx [hunk 27]
```diff
@@ -0,0 +1,179 @@
...
+```
+
+## Features
+
+- **Slider** — CSS scroll-snap, swipe on mobile, mouse drag on desktop
+- **Thumbnails** — synced via IntersectionObserver, desktop only
+- **Counter** — fraction (`1/5`), always visible
+- **Arrows** — hover reveal animation, desktop only
+- **Dialog** — fullscreen lightbox with zoom (click/wheel/double-tap) + pan
+- **Keyboard** — ArrowLeft/Right in dialog, Escape to close
```
[the remark this answers is not in the document]
Comment (spokospace) [reply src/pages/components/product-gallery.mdx:23]: Fixed. Updated the PR description — changed `role="tablist"` to `aria-pressed on thumbnail buttons` and added `prefers-reduced-motion support` to accurately reflect the current ARIA pattern.

[the remark this answers is not in the document]
Comment (spokospace) [reply src/pages/components/product-gallery.mdx:23]: Note: the earlier Copilot suggestion to move `setPointerCapture` from `pointermove` to `pointerdown` (comment from the first re-review) turned out to break click events on slides and thumbnails. Fixed in eeb3faa — `setPointerCapture` is now called in `pointermove` after the 5px drag threshold, so simple clicks still propagate to child elements.
